substrate/frame/examples/authorization-tx-extension/src/extensions.rs

```diff
@@ -0,0 +1,149 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::{fmt, marker::PhantomData};
+
+use codec::{Decode, Encode};
+use frame_support::traits::OriginTrait;
+use scale_info::TypeInfo;
+use sp_runtime::{
+	impl_tx_ext_default,
+	traits::{
+		DispatchInfoOf, IdentifyAccount, DispatchOriginOf, TransactionExtension, ValidateResult, Verify,
+	},
+	transaction_validity::{InvalidTransaction, ValidTransaction},
+};
+
+use crate::pallet_coownership::{Config, Origin};
+
+/// Helper struct to organize the data needed for signature verification of both parties involved.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, TypeInfo)]
+pub struct AuthCredentials<Signer, Signature> {
+	first: (Signer, Signature),
+	second: (Signer, Signature),
+}
+
+/// Extension that, if activated by providing a pair of signers and signatures, will authorize a
+/// coowner origin of the two signers. Both signers have to construct their signatures on all of the
+/// data that follows this extension in the `TransactionExtension` pipeline, their implications and
+/// the call. Essentially resign the transaction from this point onwards in the pipeline by using
+/// the `inherited_implication`, as shown below.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T, Signer, Signature))]
+pub struct AuthorizeCoownership<T: Config, Signer, Signature> {
+	inner: Option<AuthCredentials<Signer, Signature>>,
+	_phantom: PhantomData<T>,
+}
+
+impl<T: Config, Signer, Signature> Default for AuthorizeCoownership<T, Signer, Signature> {
+	fn default() -> Self {
+		Self { inner: None, _phantom: Default::default() }
+	}
+}
+
+impl<T: Config, Signer, Signature> AuthorizeCoownership<T, Signer, Signature> {
+	/// Creates an active extension that will try to authorize the coownership origin.
+	pub fn new(first: (Signer, Signature), second: (Signer, Signature)) -> Self {
+		Self { inner: Some(AuthCredentials { first, second }), _phantom: Default::default() }
+	}
+}
+
+impl<T: Config, Signer, Signature> fmt::Debug for AuthorizeCoownership<T, Signer, Signature> {
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "AuthorizeCoownership")
```

Comment:
It is convention. In some exceptions, if there is a primitive field like a `u64` it is displayed. In this particular example though it doesn't matter anyway since it's a demo.